lib/github.ml

```diff
@@ -235,6 +250,79 @@ let assert_tag_exists ~dry_run tag =
   if Vcs.tag_exists ~dry_run repo tag then Ok ()
   else R.error_msgf "%s is not a valid tag" tag
 
+(* Ask the user then push the tag. Guess the ssh URI from the dev-repo.
+   This function can abort:
+   - The user answered "N" to pushing the tag
+   - The push failed
+
+   This function does nothing if the tag is already present on the remote and
+   point to the same ref. *)
+let push_tag ~dry_run ~yes ~dev_repo vcs tag =
+  let remote_has_tag_uptodate () =
+    Vcs.commit_id ~dirty:false ~commit_ish:tag vcs >>= fun local_rev ->
+    Vcs.ls_remote ~dry_run vcs ~kind:`Tag ~filter:tag dev_repo >>= function
+    | [] -> Ok false
+    | (remote_rev_unpeeled, _) :: _ -> (
+        (* Resolve again in case of annotated commits (most common case).
+           This is a no-op for non-annotated commits. In case of error, we
```

Comment:
```suggestion
        (* Resolve again in case of annotated tags (most common case).
           This is a no-op for non-annotated tags. In case of error, we
```

Reply:
Multiline suggestion !